### PR TITLE
add postgresql-16.2

### PR DESCRIPTION
**Pull request by AndWac**

### CODE REVIEW

File: make-rules/shared-macros.mk [hunk 1]
```diff
@@ -906,6 +906,7 @@ PKG_MACROS +=   PERL_VERSION=$(PERL_VERSION)
 # |         12         |   2024-11-14   |
 # |         14         |   2026-11-12   |
 # |         15         |   2027-11-11   |
+# |         16         |   2028-11-09   |
 # +--------------------+----------------+
 #
 # See https://www.postgresql.org/support/versioning/
```
Comment (mtelka) [inline make-rules/shared-macros.mk:912]: Please update `PG_VERSION` below too.

Comment (AndWac) [reply make-rules/shared-macros.mk:912]: I want to wait until some users had the chance to test the new version.

**Comment by mtelka** [comment]
* Test results are same as for version 15 (`timestamptz` failed, everything else passed) so 16 should not be worse than 15.
* The default version will be set to 16 anyway once this is installed (see mediator).
* `PG_VERSION` is for other components only (when they are rebuild), so it will take effect slowly.

In short, no need to wait.

**Comment by mtelka** [comment]
Great. Thank you!

**Pull request merged**